src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Entry<K, V>[]) new Entry[INITIAL_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Entry<K, V> entry = table[index];
+        while (entry != null) {
+            if ((key == null && entry.key == null) || (key != null && key.equals(entry.key))) {
```

Comment:
Condition may be simplified in all similar cases:
```suggestion
            if ((key == entry.key) || (key != null && key.equals(entry.key))) {
```